docs/fundamentals/code-analysis/quality-rules/ca1062.md

```diff
@@ -82,30 +82,52 @@ Examples:
 |`dotnet_code_quality.CA1062.null_check_validation_methods = NS.MyType.Validate(ParamType)` | Matches specific method 'Validate' with given fully qualified signature
 |`dotnet_code_quality.CA1062.null_check_validation_methods = NS1.MyType1.Validate1(ParamType)|NS2.MyType2.Validate2(ParamType)` | Matches specific methods 'Validate1' and 'Validate2' with respective fully qualified signature
 
-### Excluded symbol names
+### Include or exclude selected APIs
```

Comment:
I am sorry, I did a search and replace without at all paying attention to the context of the rule. Let me get back to them all and ping you when I am over.

Reply:
I read your comment too quickly yesterday. I don't know what to answer, basically all configurabilities are around tweaking the code on which the analysis should run. We could rename `Configurability` into `Include or exclude selected APIs` but it feels a little weird. Besides `excluded_symbol_names` and `excluded_type_names_with_derived_types` always work in pair (we currently always implement both) so it seems better to have them grouped.

Let me know what you think is best for the structure and I will redo all the changes.

Tagging @mavasani for his opinion here.

Reply:
Given that we now have a large number of options that are shared across multiple rules, and their description + examples is getting quite verbose, I wonder if we should do the following refactor:

1. For options that are applicable to a single rule, add the option description + editorconfig example in the rule doc page itself under a Configuration section (just like we do today)
2. For options are applicable to multiple rules, move the option description + editorconfig example to a separate rule agnostic doc page, such that each rule doc page just has a single sentence for the option with a hyperlink to this common rule agnostic doc page section. The rule agnostic doc page can have an additional Configurable rules section (similar to https://github.com/dotnet/roslyn-analyzers/blob/master/docs/Analyzer%20Configuration.md#analyzed-api-surface).

Alternatively, if @gewarren is fine, we can consider completely removing the option configuration description + examples from individual CA rule doc pages in this repo and have a single description line for each configurable option that directly hyperlinks to relevant section in https://github.com/dotnet/roslyn-analyzers/blob/master/docs/Analyzer%20Configuration.md, which should always be up-to-date. I think this will probably ensure the least amount of duplicate docs.

Reply:
@gewarren What do you suggest?

Reply:
I have the same concern about so much duplicated info. I'd actually suggest a third option: create an "include" file for each option that applies to multiple rules and put it in [this folder](https://github.com/dotnet/docs/tree/master/includes/code-analysis). I'd recommend putting an H3 heading with the title of the option (e.g. `### Exclude specific symbols`) at the top of the include file. Then add a references to the include files from each applicable rule. (For examples of using include files, see https://raw.githubusercontent.com/dotnet/docs/master/docs/fundamentals/code-analysis/quality-rules/ca2300.md.)

Reply:
Alright let me try that solution :) I will ping you tomorrow.